tests/unittests/MLTest.cpp

```diff
@@ -144,46 +163,51 @@ TEST_P(MLTest, simpleRegression) {
 
   Tensor inputs(ElemKind::FloatTy, {1, numInputs});
   Tensor expected(ElemKind::FloatTy, {1, numInputs});
-
-  auto &mod = EE_.getModule();
-  Function *F = mod.createFunction("simpleRegression");
-  Placeholder *A =
-      mod.createPlaceholder(ElemKind::FloatTy, {1, numInputs}, "A", false);
-  Placeholder *Ex =
-      mod.createPlaceholder(ElemKind::FloatTy, {1, numInputs}, "E", false);
-  Node *O = F->createFullyConnected(bindings, "fc", A, 4);
-  O = F->createRELU("relu", O);
-  O = F->createRegression("reg", O, Ex);
-  auto *result = F->createSave("result", O);
-
-  bindings.allocate(A);
-  bindings.allocate(Ex);
-  auto *res = bindings.allocate(result->getPlaceholder());
+  Placeholder *A, *Ex;
+  Function *F;
+  for (auto EE : engines_) {
+
+    auto &mod = EE->getModule();
+    F = mod.createFunction("simpleRegression");
+    A = mod.createPlaceholder(ElemKind::FloatTy, {1, numInputs}, "A", false);
+    Ex = mod.createPlaceholder(ElemKind::FloatTy, {1, numInputs}, "E", false);
+    Node *O = F->createFullyConnected(bindings, "fc", A, 4);
```

Comment:
Can we use `trainingBindings` here and remove `bindings` entirely? (Same goes for all other places we have this extra `bindings`)